main.py

```diff
@@ -80,116 +75,157 @@ def __init__(self, profile_list, serial_port, servo, camera, RFID_reader):
 	# -1 is returned.
 	def searchForProfile(self, RFID):
 
-		# Search profile_list for AnimalProfile whose ID matches RFID
 		for profile in self.profile_list:
 
-			if profile.ID == RFID:
-				return profile
-			elif profile == self.profile_list[len(self.profile_list) - 1]:
-				return -1
-
-
-        # This function starts an experiment session for the animal identified in the supplied <profile>. 
-        # 
-        # Note: (TODO) Since the "RFID-proximity-pin polling" and camera recording functions are on the same thread,
-        # we currently have to interrupt the camera recording in order to poll the proximity pin to check if the
-        # session should be terminated. Obviously, that sucks. The RFID proximity polling needs to be on it's own thread,
-        # so that when it detects the RFID tag leaving, it can pass a message to the experiment session thread to interrupt it
-        # and terminate the session. This function will need to be redesigned to facilitate this.
-	def startSession(self, profile):
-
-
-		session_start_time = time.time()	
-		subsession_counter = 0
-
-		while True:
+		    if profile.ID == RFID:
+                        return profile
 
-                        # If RFID proximity pin returns high, then RFID tag is still present. Continue session. Else, stop session.
-			if self.RFID_reader.readProximityState()
+                return -1
 
-                            # Raise pellet arm
-			    self.servo.setAngle(10, 90)
 
-                            #TODO: <video_output_path> should not be constructed by startSession(). This path should be supplied by <AnimalProfile.profile>.
-			    video_output_path = profile.session_history_directory +"/Videos/" + str(profile.ID) + "_session#_"  + str(profile.session_count) + "." + str(while_counter) + ".avi"
+        # This function starts an experiment session for the animal identified in the supplied <profile>.
+        # A session is only started if the IR beam is broken and only continues while this remains true.
+        # Each session forks a process that records video for the duration of the session and forks
+        # another process which controls the servo for the duration of the session. Each session
+        # will also log data about itself. As soon as the session detects the IR beam reconnect,
+        # a destruct signal will be sent to all forked processes and this function will wait to join
+        # with those processes before concluding the session.
+        #TODO: This function seems really bloated. Try to break it down into smaller pieces.
+	def startSession(self, profile):
```

Comment:
I would break this out a bit, this function does a lot in its current state. The bits that are good candidates for breaking out:
1. Start camera recording and return the queue object
1. Do whatever servo stuff and return the queue object
1. Wait until the beam is unbroken again
1. Do whatever teardown of the camera and servo processes

Reply:
At a high level the function could look something like:
```python
def start_session(self, profile):
    path = build_output_path(profile)
    camera_queue = start_camera_session(path)
    servo_queue = start_servo_session(path)
    wait_for_session_end()
    cleanup(camera_queue, servo_queue, path)
```